src/middlewared/middlewared/plugins/interface/link_address.py

```diff
@@ -100,9 +100,16 @@ def get_name(self, i):
 
 
 class RealInterfaceCollection(InterfaceCollection):
-    @property
-    def by_link_address(self):
-        return {i["state"]["hardware_link_address"]: i for i in self.interfaces}
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self.by_link_address = {}
+        for i in self.interfaces:
+            link_address = i["state"]["hardware_link_address"]
+            if link_address in self.by_link_address:
+                raise ValueError(f"Interfaces {self.by_link_address[link_address]['name']!r} and {i['name']!r} have "
```

Comment:
Let's raise a custom exception class and catch it specifically and log a message that caters to this scenario.